.github/workflows/test-docs.yml

```
@@ -36,7 +36,7 @@ jobs:
           ENDOFFILE
       - if: success() || failure()
         name: Upload JUnit Test Results
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
```

this will not work, let's stick to the v3 version
```suggestion
        uses: actions/upload-artifact@v3
```